test/e2e_node/topology_manager_test.go

```diff
@@ -350,27 +499,231 @@ func runTopologyManagerSuiteTests(f *framework.Framework) {
+
+	ctnAttrs := []tmCtnAttribute{
+		{
+			ctnName:       "gu-container",
+			cpuRequest:    cpuAmount,
+			cpuLimit:      cpuAmount,
+			deviceName:    sriovResourceName,
+			deviceRequest: deviceAmount,
+			deviceLimit:   deviceAmount,
+		},
+	}
+
+	podName := "gu-pod"
+	framework.Logf("creating pod %s attrs %v", podName, ctnAttrs)
+	pod := makeTopologyManagerTestPod(podName, numalignCmd, ctnAttrs)
+
+	pod = f.PodClient().Create(pod)
+	err := e2epod.WaitForPodCondition(f.ClientSet, f.Namespace.Name, pod.Name, "Failed", 30*time.Second, func(pod *v1.Pod) (bool, error) {
+		if pod.Status.Phase != v1.PodPending {
+			return true, nil
+		}
+		return false, nil
+	})
+	framework.ExpectNoError(err)
+	pod, err = f.PodClient().Get(pod.Name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+
+	if pod.Status.Phase != v1.PodFailed {
+		framework.Failf("pod %s not failed: %v", pod.Name, pod.Status)
+	}
+	if pod.Status.Reason != topologyError {
+		framework.Failf("pod %s failed for wrong reason: %v", pod.Name, pod.Status)
+	}
+
+	deletePods(f, []string{pod.Name})
+}
+
+func getSRIOVDevicePluginConfigMap(cmFile string) *v1.ConfigMap {
+	cmData := testfiles.ReadOrDie(SRIOVDevicePluginCMYAML)
+	var err error
+
+	// the SRIOVDP configuration is hw-dependent, so we allow per-test-host customization.
+	framework.Logf("host-local SRIOV Device Plugin Config Map %q", cmFile)
+	if cmFile != "" {
+		cmData, err = ioutil.ReadFile(cmFile)
+		if err != nil {
+			framework.Failf("unable to load the SRIOV Device Plugin ConfigMap: %v", err)
+		}
+	} else {
+		framework.Logf("Using built-in SRIOV Device Plugin Config Map")
+	}
+
+	return readConfigMapV1OrDie(cmData)
+}
+
+func runTopologyManagerNodeAlignmentSuiteTests(f *framework.Framework, configMap *v1.ConfigMap, reservedSystemCPUs string, numaNodes, coreCount int) {
+	var err error
+
+	ginkgo.By(fmt.Sprintf("Creating configMap %v/%v", metav1.NamespaceSystem, configMap.Name))
+	if _, err = f.ClientSet.CoreV1().ConfigMaps(metav1.NamespaceSystem).Create(configMap); err != nil {
+		framework.Failf("unable to create test configMap %s: %v", configMap.Name, err)
+	}
+
+	serviceAccount := readServiceAccountV1OrDie(testfiles.ReadOrDie(SRIOVDevicePluginSAYAML))
+	ginkgo.By(fmt.Sprintf("Creating serviceAccount %v/%v", metav1.NamespaceSystem, serviceAccount.Name))
+	if _, err = f.ClientSet.CoreV1().ServiceAccounts(metav1.NamespaceSystem).Create(serviceAccount); err != nil {
+		framework.Failf("unable to create test serviceAccount %s: %v", serviceAccount.Name, err)
+	}
+
+	e2enode.WaitForNodeToBeReady(f.ClientSet, framework.TestContext.NodeName, 5*time.Minute)
+
+	dp := getSRIOVDevicePluginPod()
+	dp.Spec.NodeName = framework.TestContext.NodeName
+
+	ginkgo.By("Create SRIOV device plugin pod")
+	dpPod, err := f.ClientSet.CoreV1().Pods(metav1.NamespaceSystem).Create(dp)
+	framework.ExpectNoError(err)
+
+	sriovResourceName := ""
+	var sriovResourceAmount int64
+	ginkgo.By("Waiting for devices to become available on the local node")
```

Comment:
It may be a good idea to document, perhaps with a comment here, and in test doc code, that VFs should already be configure on the system under test. On my setup, I did not have any VFs configured (wiped out from doing other stuff), and the test spun here for 5 mins. 

Perhaps some example config/commands would be helpful. For example:

cat /sys/devices/pci0000:00/0000:00:01.0/0000:01:00.1/sriov_numvfs
echo 7 > /sys/devices/pci0000:00/0000:00:01.0/0000:01:00.1/sriov_numvfs

Reply:
Will post a PR against https://github.com/kubernetes/community/blob/master/contributors/devel/sig-node/e2e-node-tests.md - but will also add a comment here.